 #the id is used to indicate the record file
string id
---
string path